### PR TITLE
Job Improvement Project - Rebellion

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #4426

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * kRO Changelog: http://ro.gnjoy.com/news/notice/View.asp?BBSMode=10001&seq=7081

### CODE REVIEW

File: db/re/skill_cast_db.txt [hunk 1]
```diff
@@ -1668,7 +1668,7 @@
 //-- RL_MASS_SPIRAL
 2553,1000,1000,0,120000,0,2000,2000
 //-- RL_BANISHING_BUSTER
-2554,3000:2500:2000:1500:1000,0,0,0,0,2000,1000
+2554,1000,1000,0,0,0,1000,700
```
Comment (Tokeiburu) [inline db/re/skill_cast_db.txt]: The after-cast delay should be 400 ms and the skill cooldown should be 1600 ms.

File: db/re/skill_cast_db.txt [hunk 2]
```diff
@@ -1688,9 +1688,9 @@
 //-- RL_FALLEN_ANGEL
 2564,0,0,0,2000,0,0,-1
 //-- RL_R_TRIP
-2565,0,1000,0,0,0,3000:2500:2000:1500:1000,-1
+2565,0,1000,0,0,0,1000,-1 // !TODO: Confirm cooldown
```
Comment (Tokeiburu) [inline db/re/skill_cast_db.txt]: The cooldown is 2800:2600:2400:2200:2000:1800:1600:1400:1200:1000.

File: db/re/skill_cast_db.txt [hunk 3]
```diff
@@ -1688,9 +1688,9 @@
...
 //-- RL_D_TAIL
-2566,1200:1400:1600:1800:2000,2000,0,0,0,5000,-1
+2566,1200:1400:1600:1800:2000,1000,0,0,0,3500,-1
```
Comment (Tokeiburu) [inline db/re/skill_cast_db.txt]: The cast time was changed to 1200:1400:1600:1800:2000:2200:2400:2600:2800:3000.

**Comment by OptimusM** [comment]
RL_D_TAIL still need to have Crimson Maker status over the target to cast.
augmented damage are working ok.

RL_HAMMER_OF_GOD: are not doing the effect of augmented damage to targets inside the area of "marked" objetive.

**Comment by Badarosk0** [comment]
Rebellion's maximum status can reach 130.

File: db/re/skill_require_db.txt [hunk 4]
```diff
@@ -893,24 +893,24 @@
...
-2554,0,0,55:60:65:70:75,0,0,0,20,3,1,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_BANISHING_BUSTER
+2554,0,0,55:57:59:61:63:65:67:69:71:73,0,0,0,20,3,1,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_BANISHING_BUSTER
 2555,0,0,30:32:34:36:38,0,0,0,17:18:19:20:21,0,0,none,0,1,7940,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_B_TRAP
 2556,0,0,2,0,0,0,99,0,0,none,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//RL_FLICKER
 2557,0,0,50:55:60:65:70,0,0,0,20,3,1,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_S_STORM
 2558,0,0,45,0,0,0,17:18:19:20:21,0,0,none,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_E_CHAIN
 2559,0,0,5,0,0,0,17:18:19:20:21,3:4:5,1,none,SC_QD_SHOT_READY,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_QD_SHOT
 2560,0,0,10,0,0,0,17:18:19:20:21,0,0,none,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_C_MARKER
-2561,0,0,12:14:16:18:20,0,0,0,17,3,5,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_FIREDANCE
+2561,0,0,13:16:19:22:25:28:31:34:37:40,0,0,0,17,3,5,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_FIREDANCE
```
[the remark this answers is not in the document]
Comment (Tokeiburu) [reply db/re/skill_require_db.txt]: It should consume 10 bullets on kRO, as per the skill description and tested in-game:
스킬 사용 시, 10발의 총알을 소모한다.(스킬 발동 시, 최소 11발의 총알이 필요하다.)

You also need 11 bullets to actually be able to cast the skill, for... some odd reason.

**Comment by admkakaroto** [comment]
@nubspixel I can also confirm this.
Dragon Tail are not dealing any damage.

**Comment by admkakaroto** [comment]
Nice, @aleos89 :D

**Comment by Badarosk0** [comment]
Nothing continued after using Dragon Tail. No damage appears. Just for me?

**Comment by admkakaroto** [comment]
@Badarosk0 no, to me too =(

**Comment by DepTrai92** [comment]
And it's still count as melee physical
Not long range physical

File: src/map/skill.cpp [hunk 5]
```diff
@@ -2057,9 +2057,6 @@ int skill_additional_effect(struct block_list* src, struct block_list *bl, uint1
 	case RL_AM_BLAST:
 		sc_start(src,bl,SC_ANTI_M_BLAST,20 + 10 * skill_lv,skill_lv,skill_get_time2(skill_id,skill_lv));
 		break;
-	case RL_HAMMER_OF_GOD:
-		sc_start(src,bl,SC_STUN,100,skill_lv,skill_get_time2(skill_id,skill_lv));
-		break;
```
Comment (Tokeiburu) [inline src/map/skill.cpp]: Hammer of God needs a bit of a revamp. It no longer targets a random area (whether or not the target is marked); it is now always centered around the target. The skill was simplified to work like a normal area of effect skill. The implementation is a copy of RL_D_TAIL, except that it sends a clif_skill_poseffect packet instead.

File: db/re/skill_require_db.txt [hunk 6]
```diff
@@ -893,24 +893,24 @@
...
-2561,0,0,12:14:16:18:20,0,0,0,17,3,5,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_FIREDANCE
+2561,0,0,13:16:19:22:25:28:31:34:37:40,0,0,0,17,3,10,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_FIREDANCE
 2562,0,0,45:50:55:60:65,0,0,0,21,0,0,none,0,0,7664,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0				//RL_H_MINE
-2563,0,0,20:24:28:32:36,0,0,0,99,3,1,none,0,-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13201:13220:13221				//RL_P_ALTER
+2563,0,0,20:24:28:32:36,0,0,0,99,3,1,none,0,-1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13201:13220:13221	//RL_P_ALTER
 2564,0,0,10,0,0,0,17,0,0,none,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//RL_FALLEN_ANGEL
-2565,0,0,40:45:50:55:60,0,0,0,19,3,5,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0					//RL_R_TRIP
-2566,0,0,60:70:80:90:100,0,0,0,21,5,1,none,0,0,7665,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0				//RL_D_TAIL
+2565,0,0,43:46:49:52:55:58:61:64:67:70,0,0,0,19,3,5,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_R_TRIP
+2566,0,0,55:60:65:70:75:80:85:90:95:100,0,0,0,21,5,1,none,0,0,7665,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_D_TAIL
```
Comment (Tokeiburu) [inline db/re/skill_require_db.txt]: No ammunition is required, change to:
2566,0,0,55:60:65:70:75:80:85:90:95:100,0,0,0,21,0,0,none,0,0,7665,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0	//RL_D_TAIL

File: src/map/battle.cpp [hunk 7]
```diff
@@ -4339,7 +4345,7 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 		case RL_HAMMER_OF_GOD:
 			skillratio += -100 + 2800 + 1400 * skill_lv;
```
[the remark this answers is not in the document]
Comment (Tokeiburu) [reply src/map/battle.cpp]: I thought I had already commented on that, I guess I forgot ;x. The formula should be:
```
		case RL_HAMMER_OF_GOD:
			skillratio += -100 + 100 * skill_lv;
			if (sd) {
				if (wd->miscflag & 8)
					skillratio += 400 * sd->spiritball_old;
				else if (sd->spiritball_old)
					skillratio += 150 * sd->spiritball_old;
			}
			RE_LVL_DMOD(100);
			break;
```

**Comment by kyurejl** [comment]
bug: gunslinger gatling fever is not increasing +atk

**Comment by Tolimatoi** [comment]
> bug: gunslinger gatling fever is not increasing +atk


battle.cpp

		if (sc->data[SC_GATLINGFEVER]) {
			if (tstatus->size == SZ_SMALL) {
				ATK_ADD(wd->equipAtk, wd->equipAtk2, 10 * sc->data[SC_GATLINGFEVER]->val1);
			} else if (tstatus->size == SZ_MEDIUM) {
				ATK_ADD(wd->equipAtk, wd->equipAtk2, 5 * sc->data[SC_GATLINGFEVER]->val1);
			} else if (tstatus->size == SZ_BIG)
				ATK_ADD(wd->equipAtk, wd->equipAtk2, sc->data[SC_GATLINGFEVER]->val1);
		}

add additional damage to the hit, example 1000 > 1000 ~ 1100

				ATK_ADD(wd->equipAtk, wd->equipAtk2, 10 * sc->data[SC_GATLINGFEVER]->val1);

**Comment by Badarosk0** [comment]
Quick Draw Shot

"Quick Draw Shot (Tailwind Shot)",
        "Max Level: 1",
        "^CC3399Requirement: Chain Action 1^000000",
        "Skill Form: ^FF0000Offensive^000000",
        "Description: ^777777This skill can only be cast",
        "after Chain Action is activated to deal",
        "additional damage to the target. Job",
        "level affects the number of additional",
        "attacks. If there is a target afflicted",
        "with Crimson Marker within 21x21 AoE,",
        "that target receives additional damage.^000000"

[SKID.RL_QD_SHOT] = {
		"퀵 드로우 샷 (순속의 탄환)",
		"MAX Lv : 1",
		"^777777습득조건 : 체인 액션 1^000000",
		"계열 : ^777777공격^000000",
		"내용 : ^777777체인 액션 발동 중, 퀵 드로우 샷을 발동시켜 대상에게 추가 공격을 하는 스킬이다.^000000",
		"^777777체인 액션 발동 중에 이 스킬을 연계기로 사용가능하다. 시전자의 JOBLV이 높을 수록 추가 공격 횟수가 증가한다.^000000",
		"^777777만약 시전자 주위 21 x 21 범위 내에 [크림즌 마커] 가 걸려있는 대상이 있다면, 그 대상 전원에게 퀵 드로우 샷 추가 공격이 가해진다.^000000"
	},

According to the description, this skill must affect single targets and targets that are marked (Crimson Marker). It is currently affecting in area.

**Comment by OptimusM** [comment]
> Quick Draw Shot
> 
> "Quick Draw Shot (Tailwind Shot)",
> "Max Level: 1",
> "^CC3399Requirement: Chain Action 1^000000",
> "Skill Form: ^FF0000Offensive^000000",
> "Description: ^777777This skill can only be cast",
> "after Chain Action is activated to deal",
> "additional damage to the target. Job",
> "level affects the number of additional",
> "attacks. If there is a target afflicted",
> "with Crimson Marker within 21x21 AoE,",
> "that target receives additional damage.^000000"
> 
> kRO description 01/23/20
> [SKID.RL_QD_SHOT] = {
> "퀵 드로우 샷 (순속의 탄환)",
> "MAX Lv : 1",
> "^777777습득조건 : 체인 액션 1^000000",
> "계열 : ^777777공격^000000",
> "내용 : ^777777체인 액션 발동 중, 퀵 드로우 샷을 발동시켜 대상에게 추가 공격을 하는 스킬이다.^000000",
> "^777777체인 액션 발동 중에 이 스킬을 연계기로 사용가능하다. 시전자의 JOBLV이 높을 수록 추가 공격 횟수가 증가한다.^000000",
> "^777777만약 시전자 주위 21 x 21 범위 내에 [크림즌 마커] 가 걸려있는 대상이 있다면, 그 대상 전원에게 퀵 드로우 샷 추가 공격이 가해진다.^000000"
> },
> 
> According to the description, this skill must affect single targets and targets that are marked (Crimson Marker). It is currently affecting in area.

i can confirm this issue, i just noticed it.

It´s currently adding +damage to the marked targets but whit no mark it´s working on a 21x21 area

**Comment by tumemmac** [comment]
Dragon Tail

- Skill will deal additional damage against target marked by Crimson Marker and this additional damage also affects nearby target too.

**Comment by erosama28** [comment]
Hit Barrel should not multiple the attack by the modifier instead should add the modifier at the final calculation. this makes the skill incredibly overpowered.

**Pull request merged**